packages/mui-material/src/TextField/TextField.d.ts

```diff
@@ -270,4 +277,12 @@ export type TextFieldProps = StandardTextFieldProps | FilledTextFieldProps | Out
  * - [TextField API](https://mui.com/material-ui/api/text-field/)
  * - inherits [FormControl API](https://mui.com/material-ui/api/form-control/)
  */
-export default function TextField(props: TextFieldProps): JSX.Element;
+export default function TextField<V extends TextFieldVariants>(
```

Comment:
```suggestion
export default function TextField<Variant extends TextFieldVariants>(
```

Reply:
resolved in this commit https://github.com/mui/material-ui/pull/36737/commits/8ec10da07204a02d5bee69467a7f220cbfad4f4f